Fix dependency on `pathfinder_simd` for Mac

Current code can't compile on Apple Silicon due to the dependency on `pathfinder_simd`.

Change the code to make it compatible with Apple Silicon.

EDIT: this error manifests on Rust nightly (1.82) for the time being.

```
error[E0635]: unknown feature `stdarch_arm_crc32`
  --> /Users/xxxx/.cargo/registry/src/index.crates.io-6f17d22bba15001f/pathfinder_simd-0.5.3/src/lib.rs:13:69
   |
13 | ...f_rustc_nightly), feature(stdarch_arm_crc32))]
   |                              ^^^^^^^^^^^^^^^^^

For more information about this error, try `rustc --explain E0635`.
error: could not compile `pathfinder_simd` (lib) due to 1 previous error
```

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -31,6 +31,36 @@ jobs:
     - name: Run tests
       run: cargo test --locked
 
+  build_nightly:
+    name: Test Nightly
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest]
+    runs-on: ${{ matrix.os }}
+    steps:
+    - name: Checkout source
+      uses: actions/checkout@v4
+    - name: Rust cache
+      uses: Swatinem/rust-cache@v2
+      with:
+        shared-key: "build-cache-nightly"
+        cache-all-crates: "true"
+        cache-directories: |
+          ~/.cargo/registry/index/
+          ~/.cargo/registry/cache/
+          ~/.cargo/git/db/
+          target/
+    - name: Install rust nightly
+      uses: actions-rs/toolchain@v1
+      with:
+        toolchain: nightly
+        profile: minimal
+        override: true
+    - name: Build
+      run: cargo build --locked
+    - name: Run tests
+      run: cargo test --locked
+
   clippy:
     name: Lint
     runs-on: ubuntu-latest

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -9,7 +9,10 @@ env:
 jobs:
   build:
     name: Test
-    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest]
+    runs-on: ${{ matrix.os }}
     steps:
     - name: Checkout source
       uses: actions/checkout@v4

--- a/.gitignore
+++ b/.gitignore
@@ -19,3 +19,6 @@ rust-toolchain.toml
 
 # Avoid pushing statistics report from CLI output
 *.log
+
+# Mac OS DS_Store file
+.DS_Store

